--- exercises/070_comptime5.zig
+++ answers/070_comptime5.zig
@@ -25,12 +25,12 @@
     location_x: i32 = 0,
     location_y: i32 = 0,
 
-    fn waddle(self: *Duck, x: i16, y: i16) void {
+    pub fn waddle(self: *Duck, x: i16, y: i16) void {
         self.location_x += x;
         self.location_y += y;
     }
 
-    fn quack(self: Duck) void {
+    pub fn quack(self: Duck) void {
         if (self.loudness < 4) {
             print("\"Quack.\" ", .{});
         } else {
@@ -44,12 +44,12 @@
     location_x: i32 = 0,
     location_y: i32 = 0,
 
-    fn waddle(self: *RubberDuck, x: i16, y: i16) void {
+    pub fn waddle(self: *RubberDuck, x: i16, y: i16) void {
         self.location_x += x;
         self.location_y += y;
     }
 
-    fn quack(self: RubberDuck) void {
+    pub fn quack(self: RubberDuck) void {
         // Assigning an expression to '_' allows us to safely
         // "use" the value while also ignoring it.
         _ = self;
@@ -123,8 +123,8 @@
     // Please make sure MyType has both waddle() and quack()
     // methods:
     const MyType = @TypeOf(possible_duck);
-    const walks_like_duck = ???;
-    const quacks_like_duck = ???;
+    const walks_like_duck = @hasDecl(MyType, "waddle");
+    const quacks_like_duck = @hasDecl(MyType, "quack");
 
     const is_duck = walks_like_duck and quacks_like_duck;
 
