--- exercises/110_files2.zig
+++ answers/110_files2.zig
@@ -1,6 +1,6 @@
 //
 // Prerequisite :
-//    - exercise/106_files.zig, or
+//    - exercise/109_files.zig, or
 //    - create a file {project_root}/output/zigling.txt
 //      with content `It's zigling time!`(18 bytes total)
 //
@@ -28,7 +28,7 @@
     // Get the current working directory
     const cwd = std.Io.Dir.cwd();
 
-    // try to open ./output assuming you did your 106_files exercise
+    // try to open ./output assuming you did your 109_files exercise
     var output_dir = try cwd.openDir(io, "output", .{});
     defer output_dir.close(io);
 
@@ -39,7 +39,7 @@
     // initialize an array of u8 with all letter 'A'
     // we need to pick the size of the array, 64 seems like a good number
     // fix the initialization below
-    var content = ['A']*64;
+    var content: [64]u8 = @splat('A');
     // this should print out : `AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA`
     std.debug.print("{s}\n", .{content});
 
@@ -50,12 +50,12 @@
     // can you go here to find a way to read the content?
     // https://ziglang.org/documentation/master/std/#std.Io.Reader
     // hint: look for a method that reads into a slice
-    const bytes_read = zig_read_the_file_or_i_will_fight_you(&content);
+    const bytes_read = try reader.readSliceShort(&content);
 
     // Woah, too screamy. I know you're excited for zigling time but tone it down a bit.
     // Can you print only what we read from the file?
     std.debug.print("Successfully Read {d} bytes: {s}\n", .{
         bytes_read,
-        content, // change this line only
+        content[0..bytes_read], // change this line only
     });
 }
